scylla-cdc/src/stream_generations.rs

```diff
@@ -0,0 +1,507 @@
+use futures::stream::StreamExt;
+use scylla::batch::Consistency;
+use scylla::cql_to_rust::{FromCqlVal, FromCqlValError, FromCqlValError::BadCqlType, FromRow};
+use scylla::frame::response::result::{CqlValue, Row};
+use scylla::frame::value::{Timestamp, Value, ValueTooBig};
+use scylla::query::Query;
+use scylla::{FromRow, IntoTypedRows, Session};
+use std::error::Error;
+use std::sync::Arc;
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Ord, PartialOrd, FromRow)]
+pub struct GenerationTimestamp {
+    timestamp: chrono::Duration,
+}
+
+impl Value for GenerationTimestamp {
+    fn serialize(&self, buf: &mut Vec<u8>) -> Result<(), ValueTooBig> {
+        Timestamp(self.timestamp).serialize(buf)
+    }
+}
+
+impl FromCqlVal<CqlValue> for GenerationTimestamp {
+    fn from_cql(cql_val: CqlValue) -> Result<Self, FromCqlValError> {
+        let timestamp = cql_val.as_duration().ok_or(BadCqlType)?;
+        Ok(GenerationTimestamp { timestamp })
+    }
+}
```

Comment:
When you'll be done with the last two of my nitpicks this code will be no longer needed.